src/collect.ts

```diff
@@ -0,0 +1,33 @@
+import { Err, Ok, Result } from "./result"
+import { Just, Maybe, Nothing } from "./maybe"
+
+export const collectResult = <O, E>(results: Array<Result<O, E>>): Result<Array<O>, Array<E>> => {
+  const successes: Array<O> = []
+  const failures: Array<E> = []
+
+  for (const result of results) {
+    if (result.isOk()) {
+      successes.push(result.value() as O)
+    } else {
+      failures.push(result.value() as E)
+    }
```

Comment:
Would we be able to use `.cata()` here? Instead if having to do isOk and then type assertions. With cata the type should be built-in already

Reply:
2336276e569d8268539fd44b8acc8a8c5cee2dc1